weave-js/src/components/Panel2/PanelComp.tsx

```diff
@@ -339,7 +339,18 @@ const useSplitTransformerConfigs = (
       }),
     [updateConfig, config, childConfig]
   );
-  return {baseConfig, updateBaseConfig, childConfig, updateChildConfig};
+
+  const updateChildConfig2 = useUpdateConfig2({
+    config: childConfig,
+    updateConfig: updateChildConfig,
```

Comment:
i think you want to flow in the updateConfig2 from props if it exists?

Reply:
We actually don't want to do that here because we want to create a new `updateConfig2` that operates on the child config. The `RenderTransformerComp` actually does have `updateConfig2` defined on its props so that would cause our hook to just return that instead of what we want.

Reply:
in that case, the updateConfig2 is just discarded. It seems like we could augment the existing updateConfig2 similar to 
```
  const updateChildConfig = useCallback(
    newConfig =>
      updateConfig({
        ...config,
        childConfig: {
          ...childConfig,
          ...newConfig,
        },
      }),
    [updateConfig, config, childConfig]
  );
```

if it exists. for example:

```
const incomingUpdateConfigChild2 = useMemo(() => {
    if (updateConfig2 == null) {
       return null
    }
    // Do augmentation here.
}, [])
```

then still use `useUpdateConfig2` ... this way if the updateConfig2 pattern is already established by the time we hit this component, we don't throw it away

Reply:
Nice catch

Reply:
👍 flowed the props in and created a new updateChildConfig2 function based on updateConfig2!